contracts/tokenwrapper/src/contract.rs

```diff
@@ -0,0 +1,565 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    attr, coins, from_binary, to_binary, Addr, BankMsg, Binary, CosmosMsg, Deps, DepsMut, Env,
+    Fraction, MessageInfo, Response, StdError, StdResult, Uint128, WasmMsg,
+};
+use cw2::set_contract_version;
+
+use cw20::{Cw20ExecuteMsg, Cw20ReceiveMsg};
+use cw20_base::allowances::{
+    execute_burn_from, execute_decrease_allowance, execute_increase_allowance, execute_send_from,
+    execute_transfer_from, query_allowance,
+};
+use cw20_base::contract::{
+    execute_burn, execute_mint, execute_send, execute_transfer, query_balance, query_token_info,
+};
+
+use cw20_base::state::{MinterData, TokenInfo, TOKEN_INFO};
+
+use protocol_cosmwasm::error::ContractError;
+use protocol_cosmwasm::token_wrapper::{
+    ConfigResponse, Cw20HookMsg, ExecuteMsg, FeeFromAmountResponse, GetAmountToWrapResponse,
+    InstantiateMsg, QueryMsg, UpdateConfigMsg,
+};
+
+use crate::state::{Config, Supply, CONFIG, HISTORICAL_TOKENS, TOKENS, TOTAL_SUPPLY};
+use crate::utils::{
+    calc_fee_perc_from_string, get_amount_to_wrap, get_fee_from_amount, is_valid_address,
+    is_valid_unwrap_amount, is_valid_wrap_amount, parse_string_to_uint128,
+};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:cosmwasm-tokenwrapper";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    // store token info using cw20-base format
+    let data = TokenInfo {
+        name: msg.name,
+        symbol: msg.symbol,
+        decimals: msg.decimals,
+        total_supply: Uint128::zero(),
+        // set self as minter, so we can properly execute mint and burn
+        mint: Some(MinterData {
+            minter: env.contract.address,
+            cap: None,
+        }),
+    };
+    TOKEN_INFO.save(deps.storage, &data)?;
+
+    // set supply to 0
+    let supply = Supply::default();
+    TOTAL_SUPPLY.save(deps.storage, &supply)?;
+
+    // set config
+    let governer = match msg.governer {
```

Comment:
Fix spelling of `governer` to `governor`

Reply:
Done https://github.com/webb-tools/protocol-cosmwasm/pull/30/commits/76e2d7af747cbd94df5ab288c876b01cb238a178